packages/core/src/test/codewhisperer/commands/transformByQ.test.ts

```diff
@@ -464,4 +472,92 @@ describe('transformByQ', function () {
+        const successResponse = {
+            ok: true,
+            status: 200,
+            text: () => Promise.resolve('Success'),
+        }
+        fetchStub.returns({ response: Promise.resolve(successResponse) })
+        await uploadArtifactToS3(
+            'test.zip',
+            { uploadId: '123', uploadUrl: 'http://test.com', kmsKeyArn: 'arn' },
+            'sha256',
+            Buffer.from('test')
+        )
+        sinon.assert.calledOnce(fetchStub)
+    })
+
+    it('should retry upload on retriable error and succeed', async () => {
+        const failedResponse = {
+            ok: false,
+            status: 503,
+            text: () => Promise.resolve('Service Unavailable'),
+        }
+        const successResponse = {
+            ok: true,
+            status: 200,
+            text: () => Promise.resolve('Success'),
+        }
+        fetchStub.onFirstCall().returns({ response: Promise.resolve(failedResponse) })
+        fetchStub.onSecondCall().returns({ response: Promise.resolve(successResponse) })
+        await uploadArtifactToS3(
+            'test.zip',
+            { uploadId: '123', uploadUrl: 'http://test.com', kmsKeyArn: 'arn' },
+            'sha256',
+            Buffer.from('test')
+        )
+        sinon.assert.calledTwice(fetchStub)
+    })
+
+    it('should throw error after 3 failed upload attempts', async () => {
+        const failedResponse = {
+            ok: false,
+            status: 500,
+            text: () => Promise.resolve('Internal Server Error'),
+        }
+        fetchStub.returns({ response: Promise.resolve(failedResponse) })
+
+        let error: Error | undefined
+        try {
+            await uploadArtifactToS3(
+                'test.zip',
+                { uploadId: '123', uploadUrl: 'http://test.com', kmsKeyArn: 'arn' },
+                'sha256',
+                Buffer.from('test')
+            )
+        } catch (e) {
+            error = e as Error
+        }
+
+        sinon.assert.match(error?.message.includes('Upload failed after up to 3 attempts'), true)
+        sinon.assert.calledThrice(fetchStub)
+    })
+
+    it('should not retry upload on non-retriable error', async () => {
+        const failedResponse = {
+            ok: false,
+            status: 400,
+            text: () => Promise.resolve('Bad Request'),
+        }
+        fetchStub.onFirstCall().returns({ response: Promise.resolve(failedResponse) })
+
+        let error: Error | undefined
+        try {
+            await uploadArtifactToS3(
+                'test.zip',
+                { uploadId: '123', uploadUrl: 'http://test.com', kmsKeyArn: 'arn' },
+                'sha256',
+                Buffer.from('test')
+            )
+        } catch (e) {
+            error = e as Error
+        }
+        sinon.assert.match(
+            error?.message.includes('Upload failed with status code = 400; did not automatically retry'),
```

Comment:
Any reason not to use `assert.rejects` here (and in the other tests)? Search the codebase for usage examples.